### ISSUE TITLE
My overrides

### ISSUE THREAD

**Issue by will** [testdouble/standard#34]
I just sent in a PR for one of our apps to try out standard, and I wanted to share the overrides I made. 

## Style/ModuleFunction
This one rewrote all `extend self` with `module_function` which worked for half of my modules, but broke for the ones where I was using private methods in the module. To be honest, the introduction of `module_function` went completely past me, so I haven't seen the arguments in favor of it, but this was one of two autofixs that did break my code.

## Lint/PercentStringArray
This was the other one that broke me, the auto fix dropped all my single quotes in the percent string arrays. In CSP headers, you need to have single quotes, and this went ahead and removed them all. I use `%w[]` arrays otherwise it's a lot harder to read

``` ruby
    font_src:                  %w['self' data:],
    form_action:               %w['self'],
    style_src:                 %w['self' 'unsafe-inline'],

```

vs
``` ruby
    font_src:                  ["'self'", "data:"],
    form_action:               ["'self'"],
    style_src:                 ["'self'", "'unsafe-inline'"],

```

## Style/AccessModifierDeclarations

This one prevented making a single database table method private that we're later exposing as better named readers

``` ruby
  private :price_per_month # rubocop:disable Style/AccessModifierDeclarations

  def price_per_month_dollars
    price_per_month
  end

  def price_per_month_cents
    price_per_month * 100
  end
```

I should probably do a migration and rename the column I suppose, but you know, work. I'm not strongly advocating to remove this one, but it sometimes nice to later make a method private.

## others

The other ones were for rspec, and 2 other single spot places that aren't worth mentioning

**Comment by searls** [comment]
* 👍  Style/ModuleFunction fixed by b70d86e247e077d83b11957de52b8920ae194234
* 👍  Lint/PercentStringArray fixed by 2bd4b4280eac2a57ad3be5fa558cd32b22e12b9a
* 🤷‍♂️ Style/AccessModifierDeclarations - while I don't mind this rule preventing someone from marking public/private/protected on each and every method, I think that's a pretty unusual thing to see in Ruby codebases, so the consequence of this rule making this edge case much harder to deal with doesn't seem to be worth it. Fixed by cc6701aa38904977e8fac5d8cfbce0638d896d9f

**Comment by searls** [comment]
* 👍  Style/ModuleFunction fixed by b70d86e247e077d83b11957de52b8920ae194234
* 👍  Lint/PercentStringArray fixed by 2bd4b4280eac2a57ad3be5fa558cd32b22e12b9a
* 🤷‍♂️ Style/AccessModifierDeclarations - while I don't mind this rule preventing someone from marking public/private/protected on each and every method, I think that's a pretty unusual thing to see in Ruby codebases, so the consequence of this rule making this edge case much harder to deal with doesn't seem to be worth it. Fixed by cc6701aa38904977e8fac5d8cfbce0638d896d9f
